src/utils/apollo.ts

```diff
@@ -0,0 +1,20 @@
+import {gql} from '@apollo/client';
+import AsyncStorage from '@react-native-async-storage/async-storage';
+
+export const loginMutation = (email: string, password: string) => {
+  return gql`
+    mutation Login {
+      login(data: {email: "${email}", password: "${password}"}) {
+        token
+      }
+    }
+  `;
+};
```

Comment:
Nos projetos a gente costuma utilizar as variáveis do graphql para passar argumentos para mutation/query, da uma olhada na [documentação do apollo](https://www.apollographql.com/docs/react/data/operation-best-practices#use-graphql-variables-to-provide-arguments) e tenta alterar para usar as variáveis

Reply:
Usar variáveis do GraphQL te permite reaproveitar o mesmo texto da mutation pra qualquer requisição que vc precise fazer, independenmente dos parâmetros. Com uma função vc precisa recriar esse texto a cada interação.